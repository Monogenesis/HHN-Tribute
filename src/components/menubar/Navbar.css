:root {
    --backgroundColor: #1a2833;
    --logoColor: #12dbe2;
    --borderColor: #0e8daa;
    --buttonColor: #0e93b1;
    --navbarColor: #152028;
}

.NavbarElements {
    background: var(--navbarColor);
    height: 80px;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 1.5rem;
    border-bottom: 4px solid var(--borderColor);
}


.menubar-logo {
    color: white;
    justify-self: center;
    margin-left: 5%;
    margin-top:1%;
    padding-top: 8px;
    cursor: default;
    position: relative;
    top: 10%;
}

.fa-react {
    margin-left: 0.5rem;
    font-size: 1.6rem;
}



.nav-menu {
    display: grid;
    grid-template-columns: repeat(5, auto);
    grid-gap: 10px;
    list-style: none;
    text-align: center;
    width: 70vw;
    justify-content: right;
    margin-right: 5rem;
}
.nav-links {
    color: white;
    font-family: 'Montserrat', sans-serif;
    text-decoration: none;
    padding: 0.5rem 1rem;
}

.nav-links:hover {
    background-color: var(--logoColor);
    transition: all 0.2s ease-out;
}

.fa-bars {
    color: white;
}

.nav-links-mobile {
    display: none;
}

.menu-icon {
    display: none;
    
}

@media screen and (max-width: 960px) {
    .NavbarElements {
       position: relative; 
    }

    .menubar-logo{
        margin-top: 4%;
        margin-left: -45%;
        padding-top: 10px;
        padding-bottom: 10px;
    }

    .nav-menu {
        display: flex;
        flex-direction: column;
        width:100%;
        height: 300px;
        position: absolute;
        top: 80px;
        left: -100%;
        opacity: 1;
        transition: all 0.5s ease;
    }

    .nav-menu.active {
        background: var(--borderColor);
        left: 0;
        opacity: 1;
        transition: all 0.5s ease;
        z-index: 1;

    }

    .nav-links {
        text-align: center;
        padding: 2rem;
        width: 100%;
        display: table;
        border-bottom: 2px solid var(--navbarColor);
        
    }

    .nav-links:hover {
        background-color: black;
        border-radius: 0;
    }


    .menu-icon {
        display: block;
        position: absolute;
        top: 0;
        right: 0;
        transform: translate(-100%, 60%);
        font-size: 1.8rem;
        cursor: pointer;
    
    }

    .fa-times {
        color: white;
        font-size: 2rem;
    }
}