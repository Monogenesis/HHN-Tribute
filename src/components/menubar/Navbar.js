import React, { Component } from "react";
import { NavbarElements } from "./NavbarElements";
import "./Navbar.css";
import TransmissionIndicator from "../TransmissionIndicator/TransmissionIndicator";

class Navbar extends Component {
  state = { clicked: false };

  handleClick = () => {
    this.setState({ clicked: !this.state.clicked });
  };

  render() {
    return (
      <nav className="NavbarElements">
        <div className="menubar-logo">
          <img src={require("../../images/logo.png")} alt="logo" width="200" />
        </div>
        <TransmissionIndicator />
        <div className="menu-icon" onClick={this.handleClick}>
          <i
            className={this.state.clicked ? "fas fa-times" : "fas fa-bars"}
          ></i>
        </div>

        <ul className={this.state.clicked ? "nav-menu active" : "nav-menu"}>
          {NavbarElements.map((item, index) => {
            return (
              <li key={index}>
                <a className={item.cName} href={item.url}>
                  {item.title}
                </a>
              </li>
            );
          })}
        </ul>
      </nav>
    );
  }
}

export default Navbar;
