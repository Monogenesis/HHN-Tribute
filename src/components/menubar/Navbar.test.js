import React from "react";
import renderer from "react-test-renderer";
import Navbar from "./Navbar.js"

test("Navbar renders correctly", () => {
    const tree = renderer.create(<Navbar></Navbar>).toJSON();
expect(tree).toMatchSnapshot();
})