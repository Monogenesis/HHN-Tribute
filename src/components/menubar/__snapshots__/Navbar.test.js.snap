// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  className="NavbarElements"
>
  <div
    className="menubar-logo"
  >
    <img
      alt="logo"
      src="logo.png"
      width="200"
    />
  </div>
  <div
    className="menu-icon"
    onClick={[Function]}
  >
    <i
      className="fas fa-bars"
    />
  </div>
  <ul
    className="nav-menu"
  >
    <li>
      <a
        className="nav-links"
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        className="nav-links"
        href="/profile"
      >
        Profile
      </a>
    </li>
    <li>
      <a
        className="nav-links"
        href="/heatmap"
      >
        HeatMap
      </a>
    </li>
    <li>
      <a
        className="nav-links"
        href="/registration"
      >
        Registration
      </a>
    </li>
  </ul>
</nav>
`;
